content/api/servo-assembly/GPIO.md

```diff
@@ -2,31 +2,31 @@
 title: GPIO
 ---
 
-You can use the GPIO _(General Purpose Input/Output)_ pins to read and write individual pins of the servo assembly. These can be used for anything, from microswitches to LEDs. GPIO is only available on pins 2 to 12, because pins 0 and 1 are reserved for communication with our kit.
+You can use the GPIO _(General Purpose Input/Output)_ pins for anything, from microswitches to LEDs. GPIO is only available on pins 2 to 12, because pins 0 and 1 are reserved for communication with the rest of our kit.
 
-## Pin Mode
-GPIO pins support 4 different modes, allowing it to work in 4 different ways. A pin can only have 1 mode at a time, and certain pins aren't compatible with certain modes. These Pin modes are stored in an [enum](https://docs.python.org/3/library/enum.html) which needs to be imported to use.
+## Pin mode
+GPIO pins have four different modes. A pin can only have one mode at a time, and some pins aren't compatible with certain modes. These pin modes are represented by an [enum](https://docs.python.org/3/library/enum.html) which needs to be imported before they can be used.
 
 ```python
 from robot import PinMode
-``` 
+```
 
 {{% notice note %}}
-The input modes closely resemble those of an Arduino. More information on them can be found in [their docs](https://www.arduino.cc/en/Tutorial/DigitalPins)
+The input modes closely resemble those of an Arduino. More information on them can be found in [their docs](https://www.arduino.cc/en/Tutorial/DigitalPins).
 {{% /notice %}}
 
 
 ### `PinMode.INPUT`
-Allows reading the value from the pin
+In this mode, the digital state of the pin (whether it is high or low) can be read.
 
 ### `PinMode.INPUT_PULLUP`
-Allows reading the value from the pin using the built-in [pullup resistor](https://learn.sparkfun.com/tutorials/pull-up-resistors).
+Same as `PinMode.INPUT`, but with an internal [pull-up resistor](https://learn.sparkfun.com/tutorials/pull-up-resistors) enabled.
 
 ### `PinMode.OUTPUT_HIGH`
-Set the pin to high output. Equivalent to `PinValue.HIGH`
```

Comment:
I quite liked the idea of explicitly staging it's the same as the pin value. Any reason this was removed?

Reply:
I wasn't sure if it was intended for people to use a `PinValue` where one would normally use a `PinMode`. It would work anyway though, since both are correspond to the Python strings `"high"`/`"low"`.

If this isn't intended usage of the API, then this line made it sound like `PinMode.OUTPUT_HIGH` and `PinValue.HIGH` are interchangeable, when they shouldn't be considered to be so.

If this is intended usage, it seems like `PinValue` itself is redundant, seeing as both of its values are covered by `PinMode`.

Reply:
technically yes theyre used for roughly the same thing, I like the fact we can be distinct between a mode and a value